managed/services/inventory/inventory_metrics.go

```diff
@@ -103,6 +103,16 @@ func (i *InventoryMetrics) GetAgentMetrics(ctx context.Context) ([]Metric, error
 			return err
 		}
 
+		dbNodes, err := models.FindNodes(tx.Querier, models.NodeFilters{})
```

Comment:
Maybe we can use FindNodesByID
https://github.com/percona/pmm/blob/6425880d68e8a2c9e0db38b2837175aca791f1a5/managed/models/node_helpers.go#L117

Reply:
Not in this case - we are fetching all nodes to create a map with nodeID and nodeName to be able then to pull the nodeName for each of the agents. This is much faster compared to querying nodes one-by-one. :)